src/zeekygen/zeekygen.bif

```diff
@@ -98,3 +113,35 @@ function get_record_field_comments%(name: string%): string
 	string field = accessor.substr(i + 1);
 	return comments_to_val(d->GetFieldComments(field));
 	%}
+
+## Retrieve the declaring script associated with a record field.
+##
+## The declaring script for a field is different from the declaring script
+## of the record type itself when fields were added via redef.
+##
+## name: the name of a script-level record type and a field within it formatted
+##       like a typical record field access: "<record_type>$<field>".
+##
+## Returns: the declaring script associated with the record field.  If *name* does
+##          not point to a known script-level record type or a known field within
+##          a record type, an empty string is returned.
+##
+## .. zeek:see:: get_identifier_declaring_script
+function get_record_field_declaring_script%(name: string%): string
+	%{
+	string accessor = name->CheckString();
+	size_t i = accessor.find('$');
+
+	if ( i > accessor.size() - 2 )
```

Comment:
The idea here is to cover both "`$` is the last character" and `i == string::npos`, right?

Reply:
It's the same code as employed by `get_record_field_comments()`. Your explanation is what I'm reading, too.